lib/galaxy/tool_util/verify/asserts/__init__.py

```diff
@@ -30,9 +32,16 @@
             assertion_functions[member] = value
 
 
-def verify_assertions(data: bytes, assertion_description_list):
+def verify_assertions(data: bytes, assertion_description_list, decompress=None):
     """This function takes a list of assertions and a string to check
     these assertions against."""
+    if decompress:
+        with NamedTemporaryFile(delete=False) as tmpfh:
```

Comment:
Can you use an exit stack (https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack) to clean up after the assertions have finished running ?

Reply:
Did so. Do you think a `flush` is necessary?